include/mbedtls/ssl.h

```diff
@@ -1424,6 +1424,19 @@ struct mbedtls_ssl_config
                                      *   \c psk is not \c NULL or \c psk_opaque
                                      *   is not \c 0. */
 
+#if defined(MBEDTLS_ZERO_RTT)
+     /*!< Early data indication:
+      *   0  -- MBEDTLS_SSL_EARLY_DATA_DISABLED (for no early data), and
+      *   1  -- MBEDTLS_SSL_EARLY_DATA_ENABLED (for use early data)
+      */
+    int early_data_enabled;
+#if defined(MBEDTLS_SSL_SRV_C)
+    /* Max number of bytes of early data acceptable by the server. */
+    size_t max_early_data;
+    /* Callback function for early data processing (server only). */
+    int(*early_data_callback)(mbedtls_ssl_context*, unsigned char*, size_t);
+#endif /* MBEDTLS_SSL_SRV_C */
+#endif /* MBEDTLS_ZERO_RTT */
```

Comment:
I don't see these field get in used, should we remove them?

Reply:
`max_early_data` is used by the server to allocate buffer to hold early_data. The callback is also used by the server.

On the server-side, a user configure to enable/disable early data and set the max buffer length. `mbedtls` will allocate and free the early data buffer for the server.

On the client-side, the user manages the early data buffer and set them using `mbedtls_ssl_set_early_data()` function to provide the early data to `mbedtls`.